model/patch_lifecycle.go

```diff
@@ -425,9 +426,156 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
+		return nil
+	}
+
+	type aliasGroup struct {
+		project        string
+		status         string
+		parentAsModule string
+	}
+	aliasGroups := make(map[aliasGroup][]PatchTriggerDefinition)
+	for _, aliasName := range p.Triggers.Aliases {
+		aliasFound := false
+		var alias PatchTriggerDefinition
+		for _, alias = range projectRef.PatchTriggerAliases {
+			if alias.Alias == aliasName {
+				aliasFound = true
+				break
+			}
+		}
+		if !aliasFound {
+			// the user specified a non-existent alias
+			continue
+		}
+
+		childProjectID, err := FindIdForProject(alias.ChildProject)
+		if err != nil {
+			return errors.Wrapf(err, "can't get ID for child project '%s'", alias.ChildProject)
+		}
+
+		// group patches on project, status, parentAsModule
+		group := aliasGroup{
+			project:        childProjectID,
+			status:         alias.Status,
+			parentAsModule: alias.ParentAsModule,
+		}
+		aliasGroups[group] = append(aliasGroups[group], alias)
+	}
+
+	childPatchIDs := []string{}
+	for group, definitions := range aliasGroups {
+		if group.status == "" {
+			childPatchID, err := makePatchFromDefinitions(ctx, p, group.project, group.parentAsModule, definitions)
+			if err != nil {
+				return errors.Wrap(err, "problem creating patch for alias definitions")
+			}
+			if childPatchID != "" {
+				childPatchIDs = append(childPatchIDs, childPatchID)
+			}
+		} else {
+			if err := subscribeOnPatchOutcome(p, group.status, definitions); err != nil {
+				return errors.Wrap(err, "problem subscribing alias definitions on patch outcome")
+			}
+
+		}
+	}
+
+	return p.AppendChildPatches(childPatchIDs)
+}
+
+func makePatchFromDefinitions(ctx context.Context, p *patch.Patch, projectID, parentAsModule string, definitions []PatchTriggerDefinition) (string, error) {
+	if len(definitions) == 0 {
+		return "", nil
+	}
+
+	v, project, err := FindLatestVersionWithValidProject(projectID)
+	if err != nil {
+		return "", errors.Wrapf(err, "problem getting last known project for '%s'", projectID)
+	}
+
+	yamlBytes, err := yaml.Marshal(project)
+	if err != nil {
+		return "", errors.Wrap(err, "can't marshal child project")
+	}
+
+	matchingTasks, err := project.VariantTasksForSelectors(definitions, p.GetRequester())
+	if err != nil {
+		return "", errors.Wrap(err, "problem matching tasks to alias definitions")
+	}
+
+	user, err := user.FindOneById(p.Author)
+	if err != nil {
+		return "", errors.Wrapf(err, "can't get user '%s'", p.Author)
+	}
+	if user == nil {
+		return "", errors.New("can't find parent patch author")
+	}
+
+	patchNumber, err := user.IncPatchNumber()
+	if err != nil {
+		return "", errors.Wrapf(err, "can't increment patch number for '%s'", user.Id)
+	}
+
+	newPatchID := mgobson.NewObjectId()
+	patchDoc := &patch.Patch{
+		Id:            newPatchID,
+		Githash:       v.Revision,
+		VariantsTasks: matchingTasks,
+		Project:       projectID,
+		Author:        p.Author,
+		Description:   fmt.Sprintf("Child patch for '%s'", p.Id.Hex()),
```

Comment:
You made me question why I added a description in the first place, given the patch will not be displayed to users (i.e. it will eventually be hidden from users). I removed it.